### PR TITLE
Backend logging

### PR DESCRIPTION
**Pull request by JmScherer**
## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] My code follows the style guidelines enforced by static analysis tools.
- [x] If it is a core feature, I have added thorough tests.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] Will this be part of a product update? If yes, please write one phrase about this update.

This Pull Request adds logging functionality to the backend service. 

## Pull Request Details

[https://www.wrike.com/open.htm?id=1157633549](https://www.wrike.com/open.htm?id=1157633549)

Changes made:

- Updated the backend service to include logging
- Added a `logging.conf` file to the root of the backend service to detail the expected python logging format
- Added an `annotation_logging.py` file to the `src/core` folder to facilitate annotation logging format
- Changed `log_to_file` to be logging using the python logger

**To Review:**

- [x] Static Analysis by Reviewer
- [x] Check the new annotation logs
  To check this run the following commands:
  ``` bash
  # From the root folder Rosalution
  docker compose down
  docker system prune -a --volumes

  docker compose up --build -d
  ```
  - Go to `local.rosalution.cgds/rosalution
  - Login with `developer`
  - Click on the `Create New Analysis` button
  - Upload an analysis from `etc/fixtures/import`
  - Use a terminal and check the docker logs: `docker logs rosalution-backend-1`
- [x] All Github Actions checks have passed.

Screenshot 2023-07-26 at 12 40 22 PM

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -161,6 +161,16 @@ docker images
...
 
+### Logging
+
+Logging can be found at "./logging.conf".
+
+As of now the logging is simple and only provides the logging level and the log message. If more information is desired
+for debugging, the `logging.conf` has a more detailed logger format available. This contains the level name, function
+name where the logger was called, the line in the file where the logger was called, and the logger message.
+
+Just comment out the less detailed one and uncomment the more detailed to use.
```
Comment (SeriousHorncat) [inline backend/README.md:172]: Will the server need to be restarted to load the changes to the logger? Or will this update it live while the application is running?

File: backend/src/core/annotation.py [hunk 2]
```diff
@@ -98,46 +94,53 @@ def process_tasks(annotation_queue, genomic_unit_collection):  # pylint: disable
...
+                        logger.info(
+                            '%s exception happened %s with %s and %s', annotation_log_label(), error, genomic_unit,
+                            annotation_task
+                        )
 
-                    log_to_file("\n")
                     del annotation_task_futures[future]
 
-            log_to_file("after for loop for waiting for all of the futures to finish\n\n")
+            logger.info("%s ...after for loop for waiting for all of the futures to finish", annotation_log_label())
```
Comment (SeriousHorncat) [inline backend/src/core/annotation.py:146]: Maybe we can update this log to match closer to what the initiating log for this background task says "Processing annotation tasks queue" from https://github.com/uab-cgds-worthey/rosalution/pull/108/files#diff-9f447bc75c68f58ec057bb592e1066030b1de98a6d7c5b961cdf7b4bed582af7R64 to be like  "Processing annotation tasks queue complete", you can probably come up with something that makes a little bit more sense then my suggestion

**Review by SeriousHorncat — CHANGES_REQUESTED** [review]
Really like the creative solution you derived for building the annotations logging.  Just a few notes, and I took a look at the code base and there are a few print statements that could be logged to the logger instead.

- https://github.com/uab-cgds-worthey/rosalution/blob/1fd8d71e2ee30410348f5c262a2eaca3932869a7/backend/src/repository/genomic_unit_collection.py#L197
- https://github.com/uab-cgds-worthey/rosalution/blob/1fd8d71e2ee30410348f5c262a2eaca3932869a7/backend/src/routers/auth_router.py#L37
- https://github.com/uab-cgds-worthey/rosalution/blob/1fd8d71e2ee30410348f5c262a2eaca3932869a7/backend/src/routers/auth_router.py#L64

File: backend/README.md (hunk 1 above)
Comment (JmScherer) [reply backend/README.md:172]: Good catch, the server will need to be restarted as it won't detect changes in the `logger.conf` file. I'll update the documentation to reflect this.

File: backend/src/core/annotation.py (hunk 2 above)
Comment (JmScherer) [reply backend/src/core/annotation.py]: I just changed it to be "Processing annotation tasks queue complete".

**Review by SeriousHorncat — APPROVED** [review]

**Pull request merged**